types/host.go

```diff
@@ -75,6 +75,12 @@ type HostInfo struct {
 	Timezone          string    `json:"timezone"`            // System timezone.
 	TimezoneOffsetSec int       `json:"timezone_offset_sec"` // Timezone offset (seconds from UTC).
 	UniqueID          string    `json:"id,omitempty"`        // Unique ID of the host (optional).
+
+	// FQDNError contains any errors that were encountered while looking up the FQDN for the host.
+	// FQDN lookup is "best effort". As such, we do want to capture any errors during the lookup, but
+	// separately from other errors, so users of the FQDN can treat any lookup errors less severely
+	// (or not) than errors looking up other host information, as it makes sense for their use case.
+	FQDNError error
```

Comment:
> Is having an empty `FQDN` field sufficient for callers to indicate that it could not collect the FQDN value, or do they need to know the reason why (via the error)?

Given that the `shared.FQDN()` function does a few different things to try and determine the FQDN, I think it might be useful to know which step failed, so administrators could potentially fix things on their end.  This would especially handy when we decide to fallback to the OS-reported hostname when the FQDN lookup fails; without having some record of the reason the lookup failed, it would be difficult to tell whether the lookup failed or if we have a bug.

I like the idea of an `FQDN() (string, error)` method on the `Host` interface instead of the `FQDN` field on the `HostInfo` struct.  Let me give that a shot.  Thanks.